### PR TITLE
Convenience methods for tracks, visible_range, etc.

### PR DESCRIPTION
**Pull request by jminor**
Added `video_tracks()` and `audio_tracks()` accessors to Timeline.
Added `visible_range()` to Item, to make it easy to get the range of media used including any adjacent Transitions in a Track.
Fixed `trimmed_range()` and `duration()` so they return copies of the TimeRange or RationalTime object, preventing the caller from inadvertently modifying stuff.

Addresses #206 
Addresses #207

### CODE REVIEW

**Comment by jminor** [comment]
Also made it easier to use flatten like this:
```
flattened_track = otio.algorithms.flatten_stack(timeline.video_tracks())
```

File: opentimelineio/core/item.py [hunk 1]
```diff
@@ -101,6 +101,13 @@ def trimmed_range(self):
 
         return self.available_range()
 
+    def visible_range(self):
```
Comment (avrata) [inline opentimelineio/core/item.py]: Great! 

However would it make sense to also have clear `visible_range()` and `visible_range_in_parent()`? This seems to return one or the other depending on if there is a parent or not. 

Lets say we have a track with "[Head_Leader, Clip_A, 24frame_Transition, Clip_B]" (A and B being 100 frame clips). I could see wanting to ask Clip_B for its `visible_range()` (Lets say 101-200), `trimmed_range()` (113-200),  `visible_range_in_parent()` (268-367), and `trimmed_range_in_parent()` (280-367).. (I could totally be off by one or so in those numbers somewhere). 

As to what to return when there is no parent? `trimmed_range()` seems fine.

File: opentimelineio/schema/track.py [hunk 2]
```diff
@@ -96,6 +96,25 @@ def trimmed_range_of_child_at_index(self, index, reference_space=None):
...
 
+    def visible_range_of_child(self, child):
+        """The range of this child's media visible to its parent Track.
+        Includes handles revealed by adjacent transitions (if any)."""
+
+        result = child.trimmed_range()
+
+        index = self.index(child)
+        if index > 0:
+            before = self[index-1]
```
Comment (ssteinbach) [inline opentimelineio/schema/track.py]: I'm (hoping) you can use the new `neighbors_of` implementation as a cleaner form of this.

File: opentimelineio/core/item.py (hunk 1 above)
Comment (jminor) [reply opentimelineio/core/item.py]: Yeah, I see what you're getting at. I'm planning to add an "Anatomy of a Clip" section to the wiki diagrams that we just updated which should address this directly. The differences between all these *_range() methods are confusing without pictures, which is kind of a bug in the API.

File: opentimelineio/schema/track.py (hunk 2 above)
Comment (jminor) [reply opentimelineio/schema/track.py]: Done.

**Comment by ssteinbach** [comment]
Also addresses #211

**Comment by jminor** [comment]
@ssteinbach can you take another look? I replaced visible_range_of_child() with handles_of_child() which gets more directly at what visible_range() needs from its parent.

**Pull request merged**